### PR TITLE
offers: fix our blinded path setting in invoices

### PR DESCRIPTION
**Pull request by rustyrussell**
And fix pay to pay them properly too!

Thanks @carlaKC for the reports!
Closes: https://github.com/ElementsProject/lightning/issues/7161
Closes: https://github.com/ElementsProject/lightning/issues/7179

### CODE REVIEW

File: plugins/offers.c [hunk 1]
```diff
@@ -1227,8 +1227,12 @@ static const char *init(struct plugin *p,
 {
 	rpc_scan(p, "getinfo",
 		 take(json_out_obj(NULL, NULL, NULL)),
-		 "{id:%}", JSON_SCAN(json_to_pubkey, &id),
-		 "{blockheight:%}", JSON_SCAN(json_to_u32, &blockheight));
+		 "{id:%}", JSON_SCAN(json_to_pubkey, &id));
+
+	rpc_scan(p, "getchaininfo",
+		 take(json_out_obj(NULL, "last_height", "0")),
+		 "{headercount:%}", JSON_SCAN(json_to_u32, &blockheight));
```
Comment (vincenzopalazzo) [inline plugins/offers.c:1234]: Bloody hell! good catch!

File: plugins/offers_invreq_hook.c [hunk 2]
```diff
@@ -258,15 +259,15 @@ static u8 *create_enctlv(const tal_t *ctx,
...
+			 const struct pubkey *next_node_id,
 			 struct tlv_encrypted_data_tlv_payment_relay *payment_relay,
 			 struct tlv_encrypted_data_tlv_payment_constraints *payment_constraints,
 			 u8 *path_secret,
 			 struct pubkey *node_alias)
 {
 	struct tlv_encrypted_data_tlv *tlv = tlv_encrypted_data_tlv_new(tmpctx);
 
-	tlv->short_channel_id = next_scid;
+	tlv->next_node_id = cast_const(struct pubkey *, next_node_id);
```
Comment (vincenzopalazzo) [inline plugins/offers_invreq_hook.c:270]: I did not know about this `cast_const`, `ccan` is like the std that C should have

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/7311/commits/83cb66d49060fe87d1fdd2d37c3c6fc2645c2ea3

I pile my ACK on top of the https://github.com/ElementsProject/lightning/pull/7304 where thsi PR is based


I think by the next few days I should be able to test the iterop with ldk by running https://github.com/vincenzopalazzo/lampo.rs/blob/main/tests/tests/src/lampo_cln_tests.rs#L372

**Comment by endothermicdev** [comment]
trivial bolt quote and spelling fixup

**Comment by rustyrussell** [comment]
Trivial rebase after other merges

**Review by endothermicdev — APPROVED** [review]
ACK 2f7d79ebbc09b5482a361bc9f0deda9d83961163

**Pull request merged**